Src/BootCamp.Chapter/Simulation.cs

```diff
@@ -0,0 +1,47 @@
+﻿using BootCamp.Chapter.Schools;
+using BootCamp.Chapter.Students;
+using BootCamp.Chapter.Subjects;
+using BootCamp.Chapter.Teachers;
+using System;
+
+namespace BootCamp.Chapter
+{
+    public class Simulation
+    {
+        public static void Run()
+        {
+            ISchool<UniversityStudent> university = new University<UniversityStudent>("Massachusetts Institute of Technology", "Cambridge, Massachusetts, United States");
```

Comment:
University shouldn't accept middleSchoolStudent, but as of now it can accept them.

Reply:
Changed so schools only accept their own type of students.